sklearn/cluster/_kmeans.py

```diff
@@ -308,10 +308,16 @@ def k_means(
         - If a callable is passed, it should take arguments `X`, `n_clusters` and a
           random state and return an initialization.
 
-    n_init : int, default=10
+    n_init : 'auto' or int, default=10
         Number of time the k-means algorithm will be run with different
         centroid seeds. The final results will be the best output of
-        `n_init` consecutive runs in terms of inertia.
+        n_init consecutive runs in terms of inertia.
+
+        When `n_init='auto'`, the number of runs will be 10 if using
+        `init='random'`, and 1 if using `init='kmeans++'`
+
+        .. versionadded:: 1.1
+           Added 'auto' option for `n_init`.
```

Comment:
We need a `.. deprecated:: 1.1` here to say that the new default will change from 10 to "auto"

Reply:
We usually don't use the deprecated marker for a change of default value. Instead we add a versionchanged in 2 versions when it's effective.

Reply:
From my reading of the "Change default value of a parameter" section:

https://github.com/scikit-learn/scikit-learn/blob/45a17624bf8311cf3177dffbfc7fd2d429005846/doc/developers/contributing.rst?plain=1#L1211

It says to "docstring needs to be updated accordingly". I was under the impression we add a deprecated marker and then change it to versionchanged marker in two versions.

Reply:
to me ``deprecated`` really applies to stuff that's going to be removed. During irl discussion @lesteve proposed to set `versionchanged` in advance. I would be ok with that.

Anyway, this should be discussed in a dedicated issue. I'll submit a PR to improve the docs and be clear on what to do, where we can discuss that.

Reply:
The `.. deprecated:: 1.1` tag is on line 321 already, so I'll resolve this pending @jeremiedbb 's follow-up issue/PR.